packages/database-version-control/src/migrations/2023.01.04T15.26.58.create-application-reference-function.sql

```diff
@@ -1,43 +1,56 @@
-CREATE FUNCTION bng.fn_create_application_reference()
-   RETURNS VARCHAR(20) 
-   LANGUAGE plpgsql
+CREATE OR REPLACE FUNCTION bng.fn_create_application_reference
+  (contact_id UUID,
+	 application_type bng.APPLICATION_TYPE,
+	 number_of_retries INTEGER DEFAULT 5)
+-- Retain the original application refrence length until database structure changes
+-- for Defra ID integration are made.
+RETURNS VARCHAR(18)
+  LANGUAGE plpgsql
   AS
 $$
-DECLARE 
-	insert_application_reference VARCHAR(20);
-	insert_id UUID;
-	insert_date TIMESTAMP;
+DECLARE
+  insert_application_reference VARCHAR(20);
+  insert_id UUID;
+  first_random_string VARCHAR(4) := upper(bng.fn_random_string(4));
+  second_random_string VARCHAR(5) := upper(bng.fn_random_string(5));
```

Comment:
Are these the wrong way around? First rand string should be 5 and second 4?

This generates refs like BNGREG-SCEY-A2KBG6 rather than BNGREG-SCEY6-A2KBG

Reply:
@paulsimonandrews - As all database migrations are being rolled back before the changes in this pull request are deployed to cloud environments, the bug fix for this in the defra-id-complete branch did not propagate to this pull request during a merge. I've patched the function definition used in this pull request.